### PR TITLE
Support ability to add notes to a test

### LINKED ISSUE #402: Add ability to attach notes to test results
There are times when a test passes (or fails) but you want to draw attention to some other data about it. My specific need is to record performance numbers (my tests confirm performance is under n milliseconds but I also want to see the actual numbers).

In order to keep the interface and implementation simple, we can just piggyback the `done` argument:

```js
it('runs faster than 500ms', function (done) {
    expect(elapsed).to.be.below(500);
    done.note('Took: ' + elapsed + 'ms');
    done();
});
```

If we make `note()` return `done` we can also chain:
```js
it('runs faster than 500ms', function (done) {
    expect(elapsed).to.be.below(500);
    done.note('Took: ' + elapsed + 'ms')();
});
```

### PR DESCRIPTION
**Pull request by geek**
Closes #402

### CODE REVIEW

File: lib/runner.js [hunk 1]
```diff
@@ -628,10 +628,18 @@ internals.protect = function (item, state, callback) {
...
+        const done = (err) => {
 
             finish(err, 'done');
-        }, onCleanup);
+        };
+
+        item.notes = [];
+        done.note = (note) => {
+
+            item.notes.push(note);
```
Comment (AdriVanHoudt) [inline lib/runner.js]: why not return done as suggested in the issue? seems like a handy thing

**Comment by Marsup** [comment]
Too bad people using promises won't be able to use that.

File: README.md [hunk 2]
```diff
@@ -183,6 +183,24 @@ lab.experiment('with only', () => {
 });
 ```
 
+The `test()` callback has a `note` function attached to it that can be used to
```
Comment (hueniverse) [inline README.md]: `note()`

File: README.md [hunk 3]
```diff
@@ -183,6 +183,24 @@ lab.experiment('with only', () => {
...
+```javascript
+lab.test('attaches notes', (done) => {
+
+    Code.expect(1 + 1).to.equal(2);
+    done.note(`The current time is ${Date.now()}`);
+    done();
+});
+```
+
+Multiple notes can be appended for the same test case by simply calling `note`
```
Comment (hueniverse) [inline README.md]: `note()`

File: lib/runner.js (hunk 1 above)
Comment (geek) [inline lib/runner.js]: for stringing notes together?

`done().notes().notes().notes();` ?

Comment (AdriVanHoudt) [inline lib/runner.js]: also but also so you can do `done.note('Took: ' + elapsed + 'ms')();`

Comment (Marsup) [inline lib/runner.js]: Agreed, this doesn't hurt having it.

**Pull request merged**